src/array.c

```diff
@@ -204,7 +197,7 @@ JL_DLLEXPORT void jl_array_grow_end(jl_array_t *a, size_t inc)
     int isbitsunion = jl_genericmemory_isbitsunion(a->ref.mem);
     size_t newnrows = n + inc;
     if (!isbitsunion && elsz == 0) {
-        jl_genericmemory_t *newmem = jl_alloc_genericmemory(mtype, MAXINTVAL - 1);
+        jl_genericmemory_t *newmem = jl_alloc_genericmemory(mtype, MAXINTVAL - 2);
```

Comment:
Is this a separate bugfix?

Reply:
it's not a bugfix, but a tiny behavior change. As an optimization, this function grows to the maximum size Memory for an Array with elsize 0. This PR shrank this size by 1 (the math worked out a tiny bit better)